c_api/src/taichi_llvm_impl.cpp

```diff
@@ -167,6 +167,42 @@ void ti_export_cpu_memory(TiRuntime runtime,
 #endif  // TI_WITH_LLVM
 }
 
+TI_DLL_EXPORT TiMemory TI_API_CALL ti_import_cpu_memory(TiRuntime runtime,
+                                                        void *ptr,
+                                                        size_t memory_size) {
+  capi::LlvmRuntime *llvm_runtime =
+      static_cast<capi::LlvmRuntime *>((Runtime *)runtime);
+
+  auto &device = llvm_runtime->get();
+  auto &cpu_device = static_cast<taichi::lang::cpu::CpuDevice &>(device);
+
+  taichi::lang::DeviceAllocation device_alloc =
+      cpu_device.import_memory(ptr, memory_size);
+
+  // prepare memory object
+  TiMemory memory = devalloc2devmem(*llvm_runtime, device_alloc);
+
+  return memory;
+}
+
+TI_DLL_EXPORT TiMemory TI_API_CALL ti_import_cuda_memory(TiRuntime runtime,
+                                                         void *ptr,
+                                                         size_t memory_size) {
+  capi::LlvmRuntime *llvm_runtime =
```

Comment:
Wrap with `#ifdef TI_WITH_CUDA`, otherwise it's gonna fail MacOS CI

please refer to how `ti_export_cuda_memory` is implemented.